Docker container Logs prints stream format for each log line

## Describe the bug
Printing the logs from a container, the output includes 8 bytes of stream format information at the beginning of each line.

https://github.com/moby/moby/issues/8223
https://docs.docker.com/engine/api/v1.26/#operation/ContainerAttach

_Current Output:_

```
������7Status. booting: 0, online: 4, dying: 0, new_online: 0
```

## To Reproduce

```go
func TestDockerLogsOutput(t *testing.T) {
	ctx := context.Background()
	req := ContainerRequest{
		Image:      "alpine:latest",
		Cmd:        []string{"sh", "-c", "id -u"},
		WaitingFor: wait.ForExit(),
	}
	container, err := GenericContainer(ctx, GenericContainerRequest{
		ContainerRequest: req,
		Started:          true,
	})
	if err != nil {
		t.Fatal(err)
	}
	defer container.Terminate(ctx)
	r, err := container.Logs(ctx)
	if err != nil {
		t.Fatal(err)
	}
	defer r.Close()
	b, err := ioutil.ReadAll(r)
	if err != nil {
		t.Fatal(err)
	}
	assert.Equal(t, "0", string(b))
}
```

**Output:**

```
=== RUN   TestDockerLogsOutput

...

        	Error Trace:	docker_test.go:1668
        	Error:      	Not equal:
        	            	expected: "0"
        	            	actual  : "\x01\x00\x00\x00\x00\x00\x00\x020\n"

        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1,2 @@
        	            	-0
        	            	+        	            	+
        	Test:       	TestDockerLogsOutput
--- FAIL: TestDockerLogsOutput (0.98s)
FAIL
FAIL	github.com/testcontainers/testcontainers-go	1.781s
```

## Expected behavior

_Expected Output:_

```
0
```

This can be solved by making TTY configurable through a new `TTY` field in the `ContainerRequest`struct.

--- a/docker.go
+++ b/docker.go
@@ -2,6 +2,7 @@ package testcontainers
 
 import (
 	"archive/tar"
+	"bufio"
 	"bytes"
 	"context"
 	"encoding/binary"
@@ -259,11 +260,59 @@ func (c *DockerContainer) inspectContainer(ctx context.Context) (*types.Containe
 // Logs will fetch both STDOUT and STDERR from the current container. Returns a
 // ReadCloser and leaves it up to the caller to extract what it wants.
 func (c *DockerContainer) Logs(ctx context.Context) (io.ReadCloser, error) {
+
+	const streamHeaderSize = 8
+
 	options := types.ContainerLogsOptions{
 		ShowStdout: true,
 		ShowStderr: true,
 	}
-	return c.provider.client.ContainerLogs(ctx, c.ID, options)
+
+	rc, err := c.provider.client.ContainerLogs(ctx, c.ID, options)
+	if err != nil {
+		return nil, err
+	}
+
+	pr, pw := io.Pipe()
+	r := bufio.NewReader(rc)
+
+	go func() {
+		var (
+			isPrefix    = true
+			lineStarted = true
+			line        []byte
+		)
+		for err == nil {
+			line, isPrefix, err = r.ReadLine()
+
+			if lineStarted && len(line) >= streamHeaderSize {
+				line = line[streamHeaderSize:] // trim stream header
+				lineStarted = false
+			}
+			if !isPrefix {
+				lineStarted = true
+			}
+
+			_, errW := pw.Write(line)
+			if errW != nil {
+				return
+			}
+
+			if !isPrefix {
+				_, errW := pw.Write([]byte("\n"))
+				if errW != nil {
+					return
+				}
+			}
+
+			if err != nil {
+				_ = pw.CloseWithError(err)
+				return
+			}
+		}
+	}()
+
+	return pr, nil
 }
 
 // FollowOutput adds a LogConsumer to be sent logs from the container's

--- a/logconsumer_test.go
+++ b/logconsumer_test.go
@@ -3,7 +3,9 @@ package testcontainers
 import (
 	"context"
 	"fmt"
+	"io/ioutil"
 	"net/http"
+	"strings"
 	"testing"
 	"time"
 
@@ -141,6 +143,7 @@ func Test_ShouldRecognizeLogTypes(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
+	defer func() { _ = c.Terminate(ctx) }()
 
 	ep, err := c.Endpoint(ctx, "http")
 	if err != nil {
@@ -181,7 +184,6 @@ func Test_ShouldRecognizeLogTypes(t *testing.T) {
 		StdoutLog: "echo this-is-stdout\n",
 		StderrLog: "echo this-is-stderr\n",
 	}, g.LogTypes)
-	_ = c.Terminate(ctx)
 }
 
 func TestContainerLogWithErrClosed(t *testing.T) {
@@ -283,4 +285,31 @@ func TestContainerLogWithErrClosed(t *testing.T) {
 				" re-requesting logs. Instead has:\n%s", consumer.Msgs[existingLogs:],
 		)
 	}
+}
+
+func TestContainerLogsShouldBeWithoutStreamHeader(t *testing.T) {
+	ctx := context.Background()
+	req := ContainerRequest{
+		Image:      "alpine:latest",
+		Cmd:        []string{"sh", "-c", "id -u"},
+		WaitingFor: wait.ForExit(),
+	}
+	container, err := GenericContainer(ctx, GenericContainerRequest{
+		ContainerRequest: req,
+		Started:          true,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer container.Terminate(ctx)
+	r, err := container.Logs(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer r.Close()
+	b, err := ioutil.ReadAll(r)
+	if err != nil {
+		t.Fatal(err)
+	}
+	assert.Equal(t, "0", strings.TrimSpace(string(b)))
 }

